src/modules/date/current-date.provider.ts

```diff
@@ -5,10 +5,20 @@ export class CurrentDateProvider {
   getLatestDateFromTodayAnd(otherDate: Date): Date {
     const now = new Date();
 
-    if (now > otherDate) {
-      return now;
+    if (otherDate > now) {
+      return otherDate;
     }
 
-    return otherDate;
+    return now;
+  }
+
+  getEarliestDateFromTodayAnd(otherDate: Date): Date {
+    const now = new Date();
+
+    if (otherDate < now) {
+      return otherDate;
+    }
+
+    return now; // TODO APIM-74: Do we use the "date" in UTC or in British local time?
```

Comment:
We ignored these issues during Mulesoft implementation:
  * timezones
  * some API calls might fail during ACBS evening job ("calculations") run. At 19:00-20:00.
  * some API calls would fail after ACBS evening job ("calculations") run. At 20:00-24:00. Reason is that ACBS moves active date to tomorrow.
  * evening runs, so I am not sure when exactly they are running and I know just rough time 19:00-20:00.

Maybe we need support API just during office hours?

I am not sure what timezone ACBS is using for it's API and what timezone DTFS client will use for his calls.

**Reason why we set future dates to today**
UKEF supports future "start" date for deals and facilities, but ACBS doesn't. We use this "start" date for these fields in ACBS: originalEffectiveDate, targetClosingDate (date then Deal becomes active), originalApprovalDate. ACBS doesn't support  future dates for all, or maybe some, of these fields, so if date is in future we set it to today. 

Other time related issues:
  * At about 7pm (London timezone) ACBS has evening jobs, "calculations", they finish maybe at 8pm and after it ACBS moves active/posting date to next business day. So today after 8pm it will be set to 30/03/2023. After 8pm Friday it will be set to Monday, and I think it also skips bank holidays. 
  * At 23:00 (after 20:00) ACBS will accept next start date anyway, so timezone issue is not critical here.
  * Maybe we use custom field that is used to charge clients, so start date is not biggest issue. I think this field might be used for it "issueDate" (ACBS name "userDefinedDate1").
  * Some API calls needs date that is not in the past, so at the moment, in Mulesoft implementation, they are failing after evening jobs (rough time 20:00). These are all calls that involve ACBS transactions: Facility activation, Value/amount amendments, Loan creations.

Reply:
Thanks Audrius! I've created a ticket, APIM-169, for this as it's a nice-to-have if the current MuleSoft implementation doesn't handle these situations